devops/create_https_wss_endpoints_aws/main.py

```diff
@@ -0,0 +1,520 @@
+
+BASE_DOMAIN_NAME        = parse_network_config("domain_name")
+ID_DOMAIN_NAME          = BASE_DOMAIN_NAME.split('.')[0]
+array_domain_name       = []
+
+#### CREATE A COMPLETE PIPELINE ####
+dict_region_elb2arn     = defaultdict(list)
+dict_region_sslcerts    = defaultdict(list)
+dict_region_tgarn       = defaultdict(list)
+dict_region_ListenerArn = defaultdict(list)
+
+
+def create_endpoints_new_network():
+    """
+    COMPLETE PIPELINE
+    * 1/ - create SSL certificates (https, and wss) on each region
+    * 2/ - create Target Groups on each region
+    * 3/ - create ELB
+    * 4/ - create listener
+
+    * / - create entries on Route53
+    """
+
+    # 1/4 - request certificates
+    # test result: passed
+    request_ssl_certificates()
+
+    # 2/4 - create tg on each region
+    # test result: passed
+    create_target_group()
+
+    # 3/4 - create elb
+    # test result: passed
+    create_elb2()
+
+    # 4/ - create listener
+    # test result: passed
+    create_listener()
+
+    # 5/ - create rule the current listener
+    # test result: passed
+    create_rule()
+
+    # 6/ - register explorer instances into the target group
+    register_explorers()
+
+
+def register_explorers():
+    """
+    register explorer nodes into the corresponding target group
+        * register the same target into tg-https and tg-wss
+    depends on:
+        * dict_region_tgarn
+        *
+    """
+    for i in range(len(ARRAY_OF_REGIONS)):
+        region = ARRAY_OF_REGIONS[i]
+        for j in range(NUM_OF_SHARDS):
+            elbv2_client = boto3.client('elbv2', region_name=region)
+            array_of_exp_shard = parse_network_config(region+'-exp-'+str(j))
+            array_instance_id_exp = retrieve_instance_id(array_of_exp_shard)
+            # REGISTER each instance_id into the TWO target groups
+            for instance in array_instance_id_exp:
+                try:
+                    # register targets into tg-s[i]-api-pga-https
+                    resp = elbv2_client.register_targets(
+                        TargetGroupArn=dict_region_tgarn[region][j],
+                        Targets=[
+                            {
+                                'Id': instance,
+                                'Port': 9500,
+                            },
+                        ]
+                    )
+                    # register targets into tg-s[i]-api-pga-wss
+                    resp2 = elbv2_client.register_targets(
+                        TargetGroupArn=dict_region_tgarn[region][j+3],
+                        Targets=[
+                            {
+                                'Id': instance,
+                                'Port': 9800,
+                            },
+                        ]
+                    )
+                    print("--registering an explorer node into TWO target groups (tg-https and tg-wss) in region " + region)
+                    sleep(4)
+                except Exception as e:
```

Comment:
Exception is too base a class to just catch blindly on. Either except on specific exceptions like something from boto or remove the try except and let it explode